### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by shivajha**
Seeker Dashboard

### CODE REVIEW

File: client/src/components/DashboardComponent/RefferedJobCard/BootstrapCards.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <Card>
+        <Card.Header>
+          <FcClock className="clock-icon" />
+          {user.jobDate && format(new Date(user.jobDate), "dd/MM/yyyy")}
+        </Card.Header>
+        <Card.Body>
+          <Card.Title>{user.companyName}</Card.Title>
+          <Card.Text>{user.jobLocation}</Card.Text>
+          <p>
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/RefferedJobCard/BootstrapCards.js:20]: remove hard code and add data from the api response.

File: client/src/components/DashboardComponent/RefferedJobCard/FetchData.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import axios from "axios";
+
+const fetchUsers = async () => {
+  try {
+    const response = await axios.get(
+      `http://localhost:8000/api/v1/seekers/644d6ef36ff422399c1639f2`
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/RefferedJobCard/FetchData.js:6]: change this. update the URL.
See how other apis are written.

File: client/src/components/DashboardComponent/RefferedJobCard/FetchData.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    );
+    const jobIds = response.data.seeker.appliedJobList.map(
+      (appliedJob) => appliedJob.jobId
+    );
+
+    const seekerJob = await axios.post("http://localhost:8000/api/v1/jobs");
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/RefferedJobCard/FetchData.js:12]: change this. update url

File: client/src/components/DashboardComponent/SeekerJobDetails.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  const [jobDetails, setJobDetails] = useState([]);
+  const [sortColumn, setSortColumn] = useState(null);
+  const [sortDirection, setSortDirection] = useState(null);
+  const [showAll, setShowAll] = useState(false);
+
+  useEffect(() => {
+    const fetchSeeker = async () => {
+      try {
+        const response = await axios.get(
+          `http://localhost:8000/api/v1/seekers/${seekerId}`
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/SeekerJobDetails.js:24]: change this,update url

File: client/src/components/DashboardComponent/SeekerJobDetails.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        );
+        const jobIds = response.data.seeker.appliedJobList.map(
+          (appliedJob) => appliedJob.jobId
+        );
+
+        // console.log("jobid", jobIds);
+        const seekerJob = await axios.post("http://localhost:8000/api/v1/jobs");
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/SeekerJobDetails.js:31]: change this,update url

File: client/src/components/DashboardComponent/SeekerJobDetails.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+  }
+  const visibleRows = showAll ? sortedJobDetails : sortedJobDetails.slice(0, 3);
+
+  return (
+    <div>
+      <div className="AppliedJobHeader">
+        <h2 style={{ fontSize: "25px", color: "#636C86", fontWeight: "600" }}>
+          Applied Opportunities
+        </h2>
+        {/* <button onClick={() => setShowAll(!showAll)}>
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/SeekerJobDetails.js:95]: remove commented code.

File: client/src/components/DashboardComponent/sidemenu/ChangePassword.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react";
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/sidemenu/ChangePassword.js:1]: delete this file, not needed.

File: client/src/components/DashboardComponent/sidemenu/Sidemenu.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+const sidemenu = () => {
+  const handleClick = () => {
+    console.log("Button clicked!");
+  };
+  return (
+    <div>
+      <div className="side">
+        <Link to="/">
+          <img src={Logo} alt="Logo" />
+          {/* <h2 className="font-bold ml-2">HiringSala</h2> */}
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/sidemenu/Sidemenu.js:18]: remove.

File: client/src/components/DashboardComponent/sidemenu/Sidemenu.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+          <FaLaptopCode />
+          Opportunities
+        </h2>
+        <h2>
+          <ChangePassword />
+        </h2>
+        <RoundButton text="Log Out" onClick={handleClick} />
+      </div>
+
+      {/* <RoundButton text="Click me" onClick={handleClick} /> */}
```
Comment (ankitmishra03) [inline client/src/components/DashboardComponent/sidemenu/Sidemenu.js:45]: remove.

File: client/src/components/Navbar/Navbar.js [hunk 10]
```diff
@@ -3,41 +3,40 @@ import { NavLink } from "react-router-dom";
...
     </div>
     <nav>
-      <ul className='flex justify-evenly'>
+      <ul className="flex justify-evenly">
         <li>
           <ul className="text-slate-700 flex gap-x-14 justify-evenly list-disc">
             <li>
-              <NavLink to="/about">
-                  About
+              <NavLink to="/about" activeStyle>
```
Comment (ankitmishra03) [inline client/src/components/Navbar/Navbar.js:17]: remove activeStyle if not needed.

File: client/src/pages/Dashboard.css [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  overflow: auto;
+  background-color: #ffffff;
+  box-sizing: border-box;
+  padding-top: 0;
+}
+
+/* .RefferalChart {
```
Comment (ankitmishra03) [inline client/src/pages/Dashboard.css:29]: remove commented code from the whole file.

File: client/src/pages/Dashboard.js [hunk 12]
```diff
@@ -1,8 +1,44 @@
...
+import RenderUsersInCards from "../components/DashboardComponent/RefferedJobCard/RenderUsersInCards";
+import Sidemenu from "../components/DashboardComponent/sidemenu/Sidemenu";
+import FileUploader from "../components/FileUploader/FileUploader";
+import TopHeading from "../components/DashboardComponent/TopHeading/TopHeading";
+import SeekerJobDetails from "../components/DashboardComponent/SeekerJobDetails";
 
 const Dashboard = () => (
-  <div className='flex justify-center items-center text-black text-3xl h-full'>
-            Dashboard
+  // <div>
```
Comment (ankitmishra03) [inline client/src/pages/Dashboard.js:10]: remove commented code from the whole file.

File: package.json [hunk 13]
```diff
@@ -26,7 +29,9 @@
     "mongoose": "^6.10.3",
     "multer": "^1.4.5-lts.1",
     "multer-s3": "^2.10.0",
-    "styled-components": "^4.4.1",
+    "react-bootstrap": "^2.7.4",
+    "react-icons": "^4.8.0",
+    "styled-components": "^5.3.9",
```
Comment (ankitmishra03) [inline package.json:34]: use "styled-components": "^4.4.1",

**Review by ankitmishra03 — CHANGES_REQUESTED** [review]

File: client/src/components/DashboardComponent/TopHeading/TopHeading.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+          <i className="fa fa-bell">
+            <MdNotificationAdd />
+          </i>
+        </div>
+        <div className="top-profile-icon">
+          <i className="fa fa-user">
+            <CgProfile />
+          </i>
+        </div>
+        <h4>Shiva Jha</h4>
```
Comment (bhavneet2000) [inline client/src/components/DashboardComponent/TopHeading/TopHeading.js:21]: Why hardcoded?

**Pull request merged**